### PR TITLE
Heuristiccluegiver

### PR DESCRIPTION
**Pull request by nhaduong**
importing Clue error
from gameplay.engine import GameEngine: ModuleNotFoundError: No module named gameplay

Giver still contains embedding handler

Working on adding some score/assassin/civilian mechanics

### CODE REVIEW

File: codenames/clue_givers/cluegiver.py [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
-            self.neg_words = [word in board for word not in target_words]
-            self.model = self.embedding_handler.embedding
-
-''' Returns list of n=NUM_CLUES of Clues for a given group of words'''
-    def get_clues(self, group, self.neg_words, num_clues=NUM_CLUES): 
+    def __init__(self, board: [str],
+                 allIDs: [int],
+                 embeddinghandler: EmbeddingHandler):
+        self.board = board
+        self.pos_words = [board[idx] for idx, val in enumerate(allIDs) if val == 1]
```
[the remark this answers is not in the document]
Comment (nhaduong) [reply codenames/clue_givers/cluegiver.py]: target_IDs are the IDs of the pos_words, while pos_words are the actual words

File: codenames/clue_givers/cluegiver.py [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
+        #TODO consider neutral words and assassin
+        self.neg_words = [word for word in board if word not in self.pos_words]
+        #self.model = embeddinghandler.embedding
+        self.model = embeddinghandler
```
[the remark this answers is not in the document]
Comment (katiestasaski) [reply codenames/clue_givers/cluegiver.py]: Fixed

File: codenames/clue_givers/cluegiver.py [hunk 3]
```diff
@@ -49,25 +52,50 @@ def get_clues(self, group, self.neg_words, num_clues=NUM_CLUES):
...
             max_neg_cosine = np.max(neg_cosine)
             if max_neg_cosine >= min_clue_cosine:
                 continue
             clues.append((Clue(clue, group, count),np.mean(clue_cosine)))
         return clues
 
-'''List of Clues sorted by descending Cosine distance'''
-    def get_next_clue(self, game_state, score): #return list of clue objects in sorted list
+    '''List of Clues sorted by descending Cosine distance'''
+    def get_next_clue(self, game_state: [int],
```
[the remark this answers is not in the document]
Comment (katiestasaski) [reply codenames/clue_givers/cluegiver.py]: Fixed

File: codenames/clue_givers/cluegiver.py [hunk 4]
```diff
@@ -1,31 +1,34 @@
...
+        self.board = board
+        self.pos_words = [board[idx] for idx, val in enumerate(allIDs) if val == 1]
+        #TODO consider neutral words and assassin
+        self.neg_words = [word for word in board if word not in self.pos_words]
+        #self.model = embeddinghandler.embedding
+        self.model = embeddinghandler
+
+    NUM_CLUES = 10
+    ''' Returns list of n=NUM_CLUES of Clues for a given group of words'''
+    def __get_clues(self, group, neg_words, num_clues=NUM_CLUES):
```
[the remark this answers is not in the document]
Comment (katiestasaski) [reply codenames/clue_givers/cluegiver.py]: Fixed

File: codenames/clue_givers/cluegiver.py [hunk 5]
```diff
@@ -49,25 +52,50 @@ def get_clues(self, group, self.neg_words, num_clues=NUM_CLUES):
             clue_cosine = np.dot(clue_vectors[:, np.newaxis], clue_vector)
             neg_cosine = np.dot(neg_vectors[:, np.newaxis], clue_vector)
             min_clue_cosine = np.min(clue_cosine)
-            
+            print('potential clue')
+            print(clue)
```
[the remark this answers is not in the document]
Comment (katiestasaski) [reply codenames/clue_givers/cluegiver.py]: Fixed

File: codenames/clue_givers/cluegiver.py [hunk 6]
```diff
@@ -49,25 +52,50 @@ def get_clues(self, group, self.neg_words, num_clues=NUM_CLUES):
...
             clues.append((Clue(clue, group, count),np.mean(clue_cosine)))
         return clues
 
-'''List of Clues sorted by descending Cosine distance'''
-    def get_next_clue(self, game_state, score): #return list of clue objects in sorted list
+    '''List of Clues sorted by descending Cosine distance'''
+    def get_next_clue(self, game_state: [int],
+                      score: int):
+        print(self.pos_words)
+        print(self.neg_words)
```
[the remark this answers is not in the document]
Comment (katiestasaski) [reply codenames/clue_givers/cluegiver.py]: Fixed

**Pull request merged**